x-pack/plugins/ingest_pipelines/server/lib/mapper.test.ts

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { csvToIngestPipeline } from './mapper';
+import { FieldCopyAction } from '../../common/types';
+
+describe('mapper', () => {
+  describe('csvToIngestPipeline()', () => {
+    it('empty file returns empty mapping', () => {
+      expect(() => {
+        csvToIngestPipeline('', FieldCopyAction.Copy);
+      }).toThrow(new Error('Error reading file: The file provided is empty.'));
+    });
+
+    it('file parsing error for invalid csv', () => {
+      const invalidCsv = `name,number
+      one|1
+      two.2
+      fourty two,42`;
+
+      expect(() => {
+        csvToIngestPipeline(invalidCsv, FieldCopyAction.Copy);
+      }).toThrow(
+        new Error(
+          'Error reading file: An unexpected issue has occured during the processing of the file.'
+        )
+      );
+    });
+
+    it('missing the required headers errors', () => {
+      const noHeadersCsv = 'srcip,,,,source.address,Copying srcip to source.address';
+
+      expect(() => {
+        csvToIngestPipeline(noHeadersCsv, FieldCopyAction.Copy);
+      }).toThrow(
+        new Error(
+          'Missing required headers: Include [source_field, destination_field] header(s) in the CSV file.'
```

Comment:
We avoid using (s) in UI copy.  It's better to write separate messages for singular and plural:

Include source_field, destination_field headers in the CSV file.
Include the source_field header in the CSV file.

Reply:
Oops! Will update in a followup PR shortly.

Reply:
https://github.com/elastic/kibana/pull/118611

Reply:
Fixed @gchaps !